core-concepts/modules/ROOT/pages/typeql/sql-vs-typeql.adoc

```diff
@@ -0,0 +1,718 @@
+= SQL versus TypeQL guide
+:test-typeql: linear
+
+This guide helps SQL users understand TypeQL by mapping familiar SQL constructs to their TypeQL equivalents.
+TypeQL is a declarative query language for TypeDB that uses pattern matching rather than table joins.
+
+[NOTE]
+====
+TypeQL operates on a graph data model (entities, relations, and attributes) rather than tables and rows.
+This fundamental difference means that some SQL concepts map directly, while others require a different approach.
+====
+
+== Schema Setup
+
+For the examples in this guide, we'll use a simple schema:
+
+[,typeql]
+----
+#!test[schema]
+define
+  entity person, owns name, owns email, owns age;
+  attribute name, value string;
+  attribute email, value string;
+  attribute age, value integer;
+  relation employment, relates employee, relates employer;
+  person plays employment:employee;
+  entity company, owns name, plays employment:employer;
+  company owns name;
+#!test[write, commit]
+insert
+  $alice isa person, has name "Alice", has email "alice@example.com", has age 30;
+  $bob isa person, has name "Bob", has email "bob@example.com", has age 25;
+  $charlie isa person, has name "Charlie", has email "charlie@example.com", has age 35;
+  $acme isa company, has name "Acme Corp";
+  $e1 isa employment, links (employee: $alice, employer: $acme);
+----
+====
+
+== Basic Query Operations
+
+=== SELECT → match + fetch/select
+
+In SQL, `SELECT` retrieves data from tables. In TypeQL, `match` finds data by pattern, and `fetch` or `select` projects the results.
+
+**SQL:**
+[source,sql]
+----
+SELECT name, email FROM person;
+----
+
+**TypeQL:**
+
+In TypeQL, use SELECT to filter out unwanted data:
+
+[,typeql]
+----
+#!test[read, count=3]
+match
+  $p isa person, has name $name, has email $email;
+select $name, $email;	# eliminates $p
+----
+
+Special case: if you're returning data and want to format it through a `fetch` clause, it can also work as a projection:
+
+[,typeql]
+----
+#!test[read, count=3]
+match
+  $p isa person, has name $name, has email $email;
+fetch {
+  "name": $name,
+  "email": $email
+};
+----
```

Comment:
I'm tempted to include `$p.name` kind of projections here too. It might be too much new information in one place, but showing how concise we can be is probably good.